client/js/index.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { applyMiddleware, createStore, combineReducers } from 'redux'
+import { Provider } from 'react-redux'
+import { Route, Switch } from 'react-router'
+import { ConnectedRouter as Router, connectRouter, routerMiddleware } from 'connected-react-router'
+import thunk from 'redux-thunk'
+import { createLogger } from 'redux-logger'
+import createSocketIOMiddleware from 'redux-socket.io';
+import history from './history.jsx'
+
+import App from './components/app.jsx'
+
+import reducers from './reducers/index.jsx'
+
+import Modals from "./components/modals/index.jsx";
+import Nav from './components/nav.jsx';
+import Landing from "./components/landing.jsx";
+
+import socket from "./socket.jsx";
+const socketIOMiddleware = createSocketIOMiddleware(socket, "socket/", { eventName: "action"});
+
+if(typeof window !== 'undefined') {
+
+    let middlewares = applyMiddleware(thunk, routerMiddleware(history), socketIOMiddleware, createLogger())
+
+    const store = createStore(
+        connectRouter(history)(
+            combineReducers({
+                ...reducers,
+            })
+        ),
+        middlewares
+    )
+
+    /** Use the location.key property as key to force a component refresh */
```

Comment:
Why do we want to force a component refresh here?

Reply:
Actually in this case you are right that we do not want. I just have that as default so that if we have other tabs on the single page app, clicking on the same tab would reload the component so that if there is any updated information it would be received.